ENIGMAsystem/SHELL/Universal_System/estring.cpp

```diff
@@ -383,4 +393,27 @@ var string_split(const std::string &str, const std::string &delim,
   return res;
 }
 
+string directory_contents(string dname) {
+  if (string_replace_all(dname, " ", "") == "") dname = ".";
+  if (directory_exists(dname)) {
+    string rpath = filename_absolute(dname);
+    #if CURRENT_PLATFORM_ID == OS_WINDOWS
+    if (rpath.back() != '\\') rpath += "\\";
+    #else
+    if (rpath.back() != '/') rpath += "/";
+    #endif
+    string item = file_find_first(rpath + "*", fa_readonly + fa_hidden + fa_sysfile + fa_volumeid + fa_directory + fa_archive);
+    string res = rpath + item;
+    while ((file_exists(rpath + item) || directory_exists(rpath + item)) && item != "") {
+      item = file_find_next();
+      if (item != "." && item != ".." && item != "")
+        res += "\n" + rpath + item;
+    }
+    if (res == rpath + item) return "";
```

Comment:
Returning here will leave the `file_find` open. Also, why do you want to return the empty string when a directory only contains one item?